### PR TITLE
Create onMounting hook

### LINKED ISSUE #15: Research History manipulation influence and Form data
Currently data input in form fields will be lost when navigating through history. It could cause to data loss. Browsers are preserving it in some cases. Pill's implementation should be as close to default browser behavior as possible.

### PR DESCRIPTION
**Pull request by tamb**
This hook fires before every content change.  This way we can clean up any assets we may want to clean up before content changes.

### CODE REVIEW

**Comment by rumkin** [comment]
Just have reviewed the PR. Currently it has conflicts. Could you clean it up and remove anything not related to the changed behavior itself?

File: src/pill.js [hunk 1]
```diff
@@ -153,12 +155,12 @@ export default function pill(selector, options) {
...
     })
     .catch(console.error)
-
+    
     onLoading(url)
   }
 
   function onClick (e) {
-    if (e.target.nodeName !== 'A') {
+    if (e.target.nodeName !== 'A' && !e.target.dataset.pillIgnore) {
```
Comment (rumkin) [inline src/pill.js:163]: Should be removed, it has already landed in another PR.

File: readme.md [hunk 2]
```diff
@@ -162,6 +169,11 @@ will be served by browser.
 
 Determine wether previuosly loaded page should be loaded from server.
 
+
```
Comment (rumkin) [inline readme.md:172]: It conflicts with already existing PR.

File: package.json [hunk 3]
```diff
@@ -5,11 +5,12 @@
...
   "devDependencies": {
     "babel-minify": "^0.5.0",
+    "copy": "^0.3.2",
     "rollup": "^1.7.4",
     "serve": "^10.1.2"
   },
   "scripts": {
-    "start": "serve -l 8080 -n example",
+    "start": "npm run build && node copy-files.js && serve -l 8080 -n example",
     "clean": "rm -rf dist",
```
Comment (rumkin) [inline package.json:14]: Presented in PR #10

File: copy-files.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+const copy = require('copy');
+
+copy('dist/*.js', 'example', ()=> console.log('copied pill.js build into example'));
```
Comment (rumkin) [inline copy-files.js:3]: This part should be removed. It's presented in PR #10.

File: example/index.html [hunk 5]
```diff
@@ -3,27 +3,30 @@
...
+          <a href='#anchor'>Anchor</a>
+        </p>
+        <div style="height: 100vh"></div>
+        <p>
+          <a name="anchor"></a> <span>Anchor</span> <a href="/pages/a">Page A</a>
+        </p>
+      </main>
     </div>
     <!-- PAGE CONTENT END -->
     <script src='/site.js' type='module'></script>
```
Comment (rumkin) [inline example/index.html:32]: Presented in PR #9

File: example/pages/a.html [hunk 6]
```diff
@@ -2,23 +2,22 @@
...
+        <h1>Page A</h1>
       <p>
         Page A content.
       </p>
+      </main>
     </div>
-    <!-- PAGE CONTENT END -->
-    <script src='/site.js' type='module'></script>
   </body>
 </html>
```
Comment (rumkin) [inline example/pages/a.html:23]: Presented in PR #9

File: example/pages/b.html [hunk 7]
```diff
@@ -2,23 +2,23 @@
...
+        <p>
+          Page B content.
+        </p>
+      </main>
+      
     </div>
-    <!-- PAGE CONTENT END -->
-    <script src='/site.js' type='module'></script>
   </body>
 </html>
```
Comment (rumkin) [inline example/pages/b.html:24]: Presented in PR #9

File: example/site.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import pill from '/pill.js'
+
```
Comment (rumkin) [inline example/site.js:1]: Conflicts with PR #9

**Review by rumkin — CHANGES_REQUESTED** [review]

**Comment by tamb** [comment]
@rumkin Question:  Should this be `onMounting` or `onLoading`?   And the fetching of the page is `onFetch`??  What naming would you want?
Currently it would be 
`onMounting` `onLoading`

But I feel it should be `onMounting/onLoading` and `onFetch`

**Comment by rumkin** [comment]
@tamb Let's implement only `onMounting` hook in this PR. I have some doubts about onFetch interface.

**Pull request closed without merging**